CMakeLists.txt

```diff
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.15.0 FATAL_ERROR) # Configurable policies: <= CMP0094
+cmake_minimum_required(VERSION 3.17.0 FATAL_ERROR) # Configurable policies: <= CMP0102
```

Comment:
I don't quite get this comment. What part of this patch requires a new cmake?

Reply:
@arvidn

The minimum version specified for `test/CMakeLists.txt` was extremely outdated. As a result, useless noisy deprecation warnings were always printed.

`bindings/python/CMakeLists.txt` requires at least 3.17, so that was chosen as the target for all CMLs (in the commit message: `Unify minimum required versions across library, bindings and tests`). The minimum version for `test/CMakeLists.txt` could have been bumped to an earlier version just for silencing the deprecation warnings, but since we're bumping it, might as well go for consistency. The top-level CML doesn't actually require anything from 3.17 and again was done for consistency.

An additional benefit is that potential future improvements are less constrained by availability of features. If it were 100% up to me I would just bump the minimum to 3.20, since that is also available on Appveyor :).

Reply:
updating the appveyor image is non-trivial though, so bumping it to the point where that has to happen will expand the scope of this change.

Reply:
@arvidn Yes, see my most recent comments on both PRs, I have made it so that updating the appveyor image is not needed to merge the improvements in this PR.